### DISCUSSION TITLE
Adding a ByteSlices() method to StringSliceCmd

### DISCUSSION THREAD

**Discussion by iamd3vil** [redis/go-redis#1735] [Ideas]
`StringSliceCmd`.Result() returns a `[]string`. There can be a `ByteSlices()` method just like how `StringCmd` has a `Bytes()` method. Our use case is that we pull lot of items from redis and write them to an `io.Writer`. Having byte slices returned will improve performance and reduces allocations. I will send a PR if this is accepted.

cc: @knadh

**Comment by knadh** [comment]
+1 for `func (cmd *StringSliceCmd) ByteSlices()`. It can use the same underlying `utils.StringToBytes()`.

@vmihailenco what do you think?

**Reply by vmihailenco** [reply]
I am not sure about this. This looks like a micro-optimization and if you think it is worth it you can do even better without allocating a `[][]byte` for `ByteSlices`. Just copy the `StringToBytes` to your project:

```go
for _, s := range ss {
    w.Write(StringToBytes(s))
}
```

When you start thinking about such optimizations you will end up with having `StringToBytes` anyway.
